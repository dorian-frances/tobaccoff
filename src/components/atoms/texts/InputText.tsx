import React, { useState } from 'react';
import { TextInput } from 'react-native-paper';
import { StyleSheet } from 'react-native';
import { ColorsEnum } from '../../../assets/colors/colors.enum';
import { FontsEnum } from '../../../assets/fonts/fonts.enum';

type CigaretteAmountTextInputProps = {
  mode?: 'flat' | 'outlined';
  label: string;
  rightText?: string;
  getCigaretteAmount: (cigaretteAmount: string) => void;
  placeholder: string;
};

const InputText = ({
  mode = 'outlined',
  label = 'Lorem ipsum',
  rightText = '/Lorem',
  getCigaretteAmount,
  placeholder,
}: CigaretteAmountTextInputProps) => {
  const [cigaretteAmount, setCigaretteAmount] = useState('');

  return (
    <TextInput
      mode={mode}
      label={label}
      right={<TextInput.Affix text={rightText} />}
      placeholder={placeholder}
      style={styles.textInputStyle}
      onChangeText={(cigaretteAmount) => setCigaretteAmount(cigaretteAmount)}
      outlineStyle={{
        width: '100%',
        borderRadius: 100,
        borderColor: ColorsEnum.INPUT_STROKE_COLOR,
      }}
      keyboardType={'numeric'}
      onEndEditing={() => getCigaretteAmount(cigaretteAmount)}
    />
  );
};

const styles = StyleSheet.create({
  textInputStyle: {
    fontFamily: FontsEnum.MEDIUM,
    fontSize: 15,
  },
});

export default InputText;
