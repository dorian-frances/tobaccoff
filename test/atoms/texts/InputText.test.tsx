import React from 'react';
import { faker } from '@faker-js/faker';
import { render } from '@testing-library/react-native';
import InputText from '../../../src/components/atoms/texts/InputText';

jest.useFakeTimers();

describe('<InputText/>', () => {
  const fakerGetCigaretteAmount = jest.fn();
  const fakeLabelText = faker.name.firstName();
  const fakePlaceHolder = faker.name.firstName();

  it('should render correctly', () => {
    const headerText = render(
      <InputText
        getCigaretteAmount={fakerGetCigaretteAmount}
        label={fakeLabelText}
        placeholder={fakePlaceHolder}
      />
    );
    expect(headerText.toJSON).toMatchSnapshot();
  });
});
