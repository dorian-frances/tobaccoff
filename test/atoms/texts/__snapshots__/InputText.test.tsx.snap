// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<InputText/> should render correctly 1`] = `[Function]`;
